framework/src/main/java/org/checkerframework/framework/type/AnnotatedTypeMirror.java

```diff
@@ -1171,13 +1179,21 @@ public void addAnnotation(AnnotationMirror a) {
         }
 
         /**
-         * Sets the parameter types of this executable type, excluding the receiver.
+         * Sets the parameter types and updates varargsType of this executable type, excluding the
+         * receiver.
          *
+         * @param varargsType the type of varargs if the method or constructor accepts a variable
+         *     number of arguments, otherwise it should be null
          * @param params an unmodifiable list of parameter types to be captured by this method,
          *     excluding the receiver
          */
-        /*package-private*/ void setParameterTypes(List<AnnotatedTypeMirror> params) {
+        /*package-private*/ void setParameterTypes(
+                List<AnnotatedTypeMirror> params, @Nullable AnnotatedArrayType varargsType) {
             paramTypes = params;
+            // varargsType cannot be null if it has value because it is @MonotonicNonNull
```

Comment:
This comment is rather confusing. Simply refer to the field `varargType`'s comment is probably better.